core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -389,6 +445,35 @@ private boolean isEndpointGatewayEnabled(EnvironmentNetworkResponse network) {
         return Optional.ofNullable(knoxTargetGroup);
     }
 
+    private Optional<TargetGroup> setupOozieHATargetGroup(Stack stack, boolean dryRun) {
+        Optional<InstanceGroup> oozieInstanceGroup = getOozieInstanceGroup(stack);
+
+        if (oozieInstanceGroup.isPresent()) {
+            LOGGER.info("Oozie HA instance group found; enabling Oozie load balancer configuration.");
+            TargetGroup oozieTargetGroup = new TargetGroup();
+            oozieTargetGroup.setType(TargetGroupType.OOZIE);
+            oozieTargetGroup.setInstanceGroups(Set.of(oozieInstanceGroup.get()));
+            if (!dryRun) {
+                LOGGER.debug("Adding target group to Oozie HA instance group.");
+                oozieInstanceGroup.get().addTargetGroup(oozieTargetGroup);
+            } else {
+                LOGGER.debug("Dry run, skipping instance group/target group linkage for Oozie.");
+            }
+            return Optional.of(oozieTargetGroup);
+        } else {
+            LOGGER.info("Oozie HA instance group not found; not enabling Oozie load balancer configuration.");
+            return Optional.empty();
+        }
+    }
+
+    private Optional<InstanceGroup> getOozieInstanceGroup(Stack stack) {
+        Set<String> oozieGroupNames = getOozieGroups(stack);
+        return stack.getInstanceGroups().stream()
+            .filter(ig -> oozieGroupNames.contains(ig.getGroupName()))
+            .filter(ig -> ig.getNodeCount() > 1)
```

Comment:
Assuming this is the part which sets up the LB only if more than 1 node (specifically 1 HG with more than 1 node) is present.

Reply:
Yes.

Reply:
So with this we are saying that only one group can contain oozie instances?

Reply:
Yes.

Reply:
Thanks